test/cypress/integration/card/payment.spec.js

```diff
@@ -148,4 +148,32 @@ describe('Standard card payment flow', () => {
       cy.location('search').should('eq', `?confirm`)
     })
   })
+
+  describe('Secure card payment page should show error', () => {
+    it('Should setup the payment and load the page', () => {
+      cy.task('setupStubs', createPaymentChargeStubs)
+      cy.visit(`/secure/${tokenId}`)
+
+      // 1. Charge will be created using this id as a token (GET)
+      // 2. Token will be deleted (DELETE)
+      // 3. Charge will be fetched (GET)
+      // 4. Service related to charge will be fetched (GET)
+      // 5. Charge status will be updated (PUT)
+      // 6. Client will be redirected to /card_details/:chargeId (304)
+      cy.location('pathname').should('eq', `/card_details/${chargeId}`)
+    })
+
+    it('Should show error when card number is less than 11 digits', () => {
+      cy.task('setupStubs', checkCardDetailsStubs)
+
+      cy.server()
+      cy.route('POST', `/check_card/${chargeId}`).as('checkCard')
+
+      cy.get('#card-no').type('1234567890')
+      cy.get('#card-no').blur()
+
+      cy.get('#card-no').should('have.class', 'govuk-input--error')
+      cy.get('#card-no-lbl').should('contain', 'Card number is not the correct length')
```

Comment:
Ah, I didn’t realise this could actually be displayed on the page. We’d need a Welsh translation. Might be easier to reuse whatever the current message is that we have for a card not being supported.